vmdb/app/models/ext_management_system.rb

```diff
@@ -131,14 +137,12 @@ def self.create_discovered_ems(ost)
   end
 
   def self.model_name_from_emstype(emstype)
-    leaf_subclasses.each do |k|
-      return k.name if k.ems_type == emstype.downcase
-    end
-    return nil
+    model_from_emstype(emstype).try(:name)
   end
 
   def self.model_from_emstype(emstype)
-    model_name_from_emstype(emstype).constantize
+    emstype = emstype.downcase
+    ExtManagementSystem.leaf_subclasses.detect { |k| k.ems_type == emstype }
```

Comment:
I still converted `leaf_subclasses` to `leaf_descendants` in my head so I could understand this :confused: